codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/AwsProtocolUtils.java

```diff
@@ -162,4 +175,154 @@ public static void handleDecodeError(GoWriter writer, String returnExtras) {
+                    attrName, attrName,
+                    namespace.getPrefix().isPresent() ? namespace.getPrefix().get() : "", namespace.getUri()
+            );
+        }
+
+        // Traverse member shapes to get attributes
+        shape.members().stream().forEach(memberShape -> {
+            if (memberShape.hasTrait(XmlAttributeTrait.class)) {
+                writeSafeMemberAccessor(context, memberShape, inputSrc, (operand) -> {
+                    // Xml Attributes should always be string
+                    writer.write("var av string");
+                    formatMemberAsString(context, memberShape, operand, "av");
+                    writer.write("$L = append($L, smithyxml.NewAttribute($S, av))",
+                            attrName, attrName, getSerializedXMLMemberName(memberShape));
+                });
+            }
+        });
+
+        writer.openBlock("$L := smithyxml.StartElement{ ", "}", dst, () -> {
+            writer.openBlock("Name:smithyxml.Name{", "},", () -> {
+                writer.write("Local: $S,", getSerializedXMLShapeName(context, shape));
+            });
+            writer.write("Attr : $L,", attrName);
+        });
+    }
+
+    // writes formatted string for member shape. This methods is used for formatting
+    // xml attributes. Shape types string, timestamp, number and boolean can target Xml.
+    private static void formatMemberAsString(GenerationContext context, MemberShape member, String src, String dest) {
+        GoWriter writer = context.getWriter();
+        Shape target = context.getModel().expectShape(member.getTarget());
+        TimestampFormatTrait.Format format = member.getMemberTrait(context.getModel(), TimestampFormatTrait.class)
+                .map(TimestampFormatTrait::getFormat).orElse(TimestampFormatTrait.Format.DATE_TIME);
+
+        if (target.isTimestampShape() || target.hasTrait(TimestampFormatTrait.class)) {
+            writer.addUseImports(SmithyGoDependency.SMITHY_TIME);
+            switch (format) {
+                case DATE_TIME:
+                    writer.write("$L = smithytime.FormatDateTime(*$L)", dest, src);
+                    break;
+                case HTTP_DATE:
+                    writer.write("$L = smithytime.FormatHTTPDate(*$L)", dest, src);
+                    break;
+                case EPOCH_SECONDS:
+                    writer.addUseImports(SmithyGoDependency.STRCONV);
+                    writer.write("$L = strconv.FormatFloat(smithytime.FormatEpochSeconds(*$L), 'f', -1, 64)", dest, src);
+                    break;
+                case UNKNOWN:
+                    throw new CodegenException("Unknown timestamp format");
+            }
+            return;
+        }
+
+        if (target.isStringShape()) {
+            writer.addUseImports(SmithyGoDependency.SMITHY_XML);
+            writer.write("$L = smithyxml.EscapeString(*$L)", dest, src);
+            return;
+        }
```

Comment:
Shouldn't escaping be handled by the encoder utility not the serializer before the encoder value is called?